dags/analyzer_helper/discord/discord_transform_raw_data.py

```diff
@@ -0,0 +1,147 @@
+        is_bot = self.user_bot_checker(author)
+        return {
+            "author_id": engaged_user,
+            "date": data["createdDate"]["$date"],
+            "source_id": data["messageId"],
+            "metadata": {
+                "thread_id": data["threadId"],
+                "channel_id": data["channelId"],
+                "bot_activity": data["isGeneratedByWebhook"] or is_bot,
+            },
+            "actions": [],
+            "interactions": [
+                {
+                    "name": interaction_type,
+                    "users_engaged_id": [author],
+                    "type": "receiver",
+                }
+            ],
+        }
+
+    def create_emitter_interaction(
+        self, user, period, data, interaction_type, engaged_user
+    ):
+        is_bot = self.user_bot_checker(engaged_user)
+        return {
+            "author_id": user,
+            "date": period,
+            "source_id": data["messageId"],
+            "metadata": {
+                "thread_id": data["threadId"],
+                "channel_id": data["channelId"],
+                "bot_activity": data["isGeneratedByWebhook"] or is_bot,
+            },
+            "actions": [],
+            "interactions": [
+                {
+                    "name": interaction_type,
+                    "users_engaged_id": [engaged_user],
+                    "type": "emitter",
+                }
+            ],
+        }
+
+    def create_transformed_item(self, data, period, interactions):
+        is_bot = self.user_bot_checker(data["author"])
+        return {
+            "author_id": data["author"],
+            "date": period,
+            "source_id": data["messageId"],
+            "metadata": {
+                "thread_id": data["threadId"],
+                "channel_id": data["channelId"],
+                "bot_activity": data["isGeneratedByWebhook"] or is_bot,
+            },
+            "actions": [
+                {
+                    "name": "message",
+                    "type": "emitter",
+                }
+            ],
+            "interactions": interactions,
+        }
+
+    def transform(
+        self, raw_data: list, platform_id: str, period: datetime
+    ) -> list[dict]:
+        transformed_data = []
+        for data in raw_data:
+            try:
+                if "author" not in data or not data["author"]:
+                    raise ValueError("Missing 'author' in raw data")
+
+                interactions = []
+
+                if data.get("replied_user"):
+                    interactions.append(
+                        self.create_interaction(
+                            "reply", [data["replied_user"]], "emitter"
+                        )
+                    )
+                    receiver_interaction = self.create_receiver_interaction(
+                        data, "reply", data["replied_user"], data["author"]
```

Comment:
Please make sure to add the equal operator for assigning values to function parameters. It could make the code much better readable

example
```
self.create_receiver_interaction(data=data, author= .... # other things
```